apps/managedidentity/managedidentity.go

```diff
@@ -446,7 +465,32 @@ func (c Client) getTokenForRequest(req *http.Request) (accesstokens.TokenRespons
 	return r, err
 }
 
-func createIMDSAuthRequest(ctx context.Context, id ID, resource string) (*http.Request, error) {
+func createAppServiceAuthRequest(ctx context.Context, id ID, resource string, c Client) (*http.Request, error) {
+	identityEndpoint := os.Getenv(identityEndpointEnvVar)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, identityEndpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("X-IDENTITY-HEADER", os.Getenv(identityHeaderEnvVar))
+	q := req.URL.Query()
+	q.Set("api-version", appServiceAPIVersion)
+	q.Set("resource", resource)
+	switch t := id.(type) {
+	case UserAssignedClientID:
+		q.Set(miQueryParameterClientId, string(t))
+	case UserAssignedResourceID:
+		q.Set(miQueryParameterResourceId, string(t))
+	case UserAssignedObjectID:
+		q.Set(miQueryParameterObjectId, string(t))
```

Comment:
I don't see this in .net or other msal's
Is this changed for App service only?

Reply:
Yes, but it appears I'm out of date on this point, as [the docs](https://learn.microsoft.com/azure/app-service/overview-managed-identity?tabs=portal%2Chttp#rest-endpoint-reference) now state the API will accept `object_id` as an alias for `principal_id`. 🤷 I'd still make this change because Azure SDK uses, and tests, `principal_id`, and these docs have been incorrect before (i.e. if you want to keep using `object_id`, make sure you test it)

Reply:
I tested with `object_id` and it worked, would you recommend that I also test with `principal_id`?

Reply:
You should run live tests before merging in any case